source/app-development/tutorials-interactive-apps/add-custom-queue/global-static-list.rst

```diff
@@ -0,0 +1,251 @@
+     job:
+       # ... resource manager specific options here ...
+     custom:
+       queues: # this can be named anything your heart desires
+         - "queue1"
+         - "queue2"
+
+.. warning::
+
+   Be sure to click "Restart Web Server" on the Dashboard everytime you edit
+   the Cluster Configuration file for changes to take effect within the app.
+
+.. _app-development-tutorials-interactive-apps-add-custom-queue-global-static-list-add-custom-attribute:
+
+Add Custom Attribute to Form
+----------------------------
+
+We want to **replace** the ``bc_queue`` form attribute with a custom HTML
+``<select>`` element (a drop-down list of options) built up of queue/partition
+options read from the corresponding cluster configuration file.
+
+.. note::
+
+   You can read more about customizing attributes in the ``form.yml`` file for
+   Interactive Apps under the
+   :ref:`interactive-development-form-customizing-attributes` section.
+
+#. First we rename the ``form.yml`` for the Jupyter Interactive App to
+   ``form.yml.erb``:
+
+   .. code-block:: shell
+
+      mv ~/ondemand/dev/jupyter/form.yml ~/ondemand/dev/jupyter/form.yml.erb
+
+   This will cause the YAML file to be processed using the `eRuby (embedded
+   Ruby)`_ templating system, which allows us to embed Ruby_ code into the YAML
+   configuration file for flow control, variable substitution, and more.
+
+#. Next we start with the default ``form.yml`` for the Jupyter Interactive App:
+
+   .. code-block:: yaml
+
+      # ~/ondemand/dev/jupyter/form.yml.erb
+      ---
+      cluster: "my_cluster"
+      attributes:
+        modules: "python"
+        conda_extensions: "1"
+        extra_jupyter_args: ""
+      form:
+        - modules
+        - conda_extensions
+        - extra_jupyter_args
+        - bc_account
+        - bc_queue
+        - bc_num_hours
+        - bc_num_slots
+        - bc_email_on_started
+
+#. We remove the following line from this file:
+
+   .. code-block:: yaml
+      :emphasize-lines: 2
+
+      - bc_account
+      - bc_queue
+      - bc_num_hours
+
+   Now when we refresh the web page for our sandbox Jupyter App we won't see
+   the "Queue" form element anymore.
+
+#. We now add in code that reads in the list of available queues/partitions
+   from the cluster configuration file and generates a custom drop-down
+   attribute with this list of queues/partitions:
+
+   .. code-block:: yaml
+      :emphasize-lines: 2-4,11-18,24
+
+      # ~/ondemand/dev/jupyter/form.yml.erb
+      <%-
+        queues = OodAppkit.clusters[:my_cluster].custom_config[:queues]
+      -%>
+      ---
+      cluster: "my_cluster"
+      attributes:
+        modules: "python"
+        conda_extensions: "1"
+        extra_jupyter_args: ""
+        custom_queue:
+          label: Queue
+          help: Please select a queue from the drop-down.
+          widget: select
+          options:
+          <%- queues.each do |q| -%>
+            - [ "<%= q %>", "<%= q %>" ]
```

Comment:
Good time to mention https://github.com/OSC/ood_core/issues/65. I didn't add this to ingest yet, but thought it was relevant considering this example.